sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/command/ShowTablesSuite.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hive.execution.command
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.execution.command.v1
+import org.apache.spark.sql.hive.test.TestHive
+
+class ShowTablesSuite extends v1.ShowTablesSuite {
+  override def version: String = "Hive V1"
+  override def defaultUsing: String = "USING HIVE"
+  override protected val spark: SparkSession = TestHive.sparkSession
+  protected override def beforeAll(): Unit = {}
+
+  test("show Hive tables") {
```

Comment:
is it specific to hive tables? if not we should put it in the base trait.

Reply:
They are not. The difference is probably in omitting explicit `USING HIVE`.

Reply:
that's not something `SHOW TABLES` should care about. Let's simply add a test in the base trait for complex patterns and this suite just extends the base trait.